### DISCUSSION TITLE
ItemContro绑定一个动态集合，动态集合里面是不同类型，如直线，椭圆，矩形的数据，还有选中后的直线，椭圆，矩形的样式，如何实现呢？

### DISCUSSION THREAD

**Discussion by WWMMFFLLRR** [BYJRK/DotNet-Discussions#141] [WPF 开发]
ItemContro绑定一个动态集合，动态集合里面是不同类型，如直线，椭圆，矩形的数据，还有选中后的直线，椭圆，矩形的样式，如何实现呢？
![image](https://github.com/BYJRK/DotNet-Discussions/assets/37059082/ec910e6a-23bf-46bc-8e25-acd37b713463)
绑定单一直线可以做到

**Comment by BYJRK** [comment]
你需要提供多种 DataTemplate，可以参考我的视频：

[WPF中用ItemsControl控件展示集合中的数据](https://b23.tv/5l12KTe)

**Comment by WWMMFFLLRR** [comment]
你好，我需要实现的功能大概是这样的，有多组组合图形，比如2组，第一组有动态条直线和动态个矩形，椭圆（这些是控制柄，鼠标左键按下拖动会根据相应控制柄功能改变这一组图形，如平移，缩放，旋转等）。第二组也是类似图形与功能。鼠标点击第一组的元素，第一组会被整体选中，第一组所有的元素（直线和控制柄）的样式都会改变，控制柄的图形会显现出来，可以控制图形。有且只能选择一组。然后下面是我写的一个小例子。

xmal前台代码部分：
![image](https://github.com/BYJRK/DotNet-Discussions/assets/37059082/e01e332d-4b94-4360-9970-a23ea86d2ca8)
![image](https://github.com/BYJRK/DotNet-Discussions/assets/37059082/e647231a-101f-4e09-bf9a-b6b85247c842)
![image](https://github.com/BYJRK/DotNet-Discussions/assets/37059082/7d9962ce-a6e9-40ed-a926-193b02283cf4)
![image](https://github.com/BYJRK/DotNet-Discussions/assets/37059082/288fb64d-9a00-4223-8f0b-a46c3bc473cb)
用了itemcontrol的嵌套来实现，第一层是绑定的多组集合的对象，第二层是绘制层，绑定的某一组的对象，这一组里面是许多直线，矩形，椭圆组成的图形。
问题在于对于不同的控制柄应该有不同的事件，我在一个DataTemplate似乎没有好的方案去实现不同控制柄的功能。
![image](https://github.com/BYJRK/DotNet-Discussions/assets/37059082/3b2d652e-b149-4542-8cfc-0a914f53df17)
如图所示，1号控制柄是控制平移的，2,3是控制缩放的，这些鼠标事件改如何绑定呢？比如1号控制柄，鼠标进入，鼠标样式变成十字，对于2，3号控制柄鼠标进入是缩放样式。
希望大佬有空可以帮忙指导一下。

**Reply by SagapoSheepL** [reply]
说一下我的拙见，DataTemplate是按照数据类型来体现UI，所以每个图形对象应该有其对应的控制柄命令Command，我刚才搜了一下，貌似命令可以绑定鼠标事件
